ggml/src/ggml-cuda/pad_reflect_1d.cu

```diff
@@ -21,32 +21,33 @@ static __global__ void pad_reflect_1d_kernel_f32(
 
     const int64_t i3 = blockIdx.z;
     const int64_t i2 = blockIdx.y;
-    const int64_t i1 = blockIdx.x;
 
-    if (i1 >= ne01 || i2 >= ne02 || i3 >= ne03) {
+    const int64_t tile1  = blockIdx.x % ne01;     // i1
+    const int64_t tile0  = blockIdx.x / ne01;     // nth i0 tile
+    const int64_t i1     = tile1;
+    const int64_t i0     = threadIdx.x + tile0 * blockDim.x;
+    if ( i0 >= ne0 || i1 >= ne01 || i2 >= ne02 || i3 >= ne03 ) {
         return;
     }
 
     const char * src0_ptr = (const char *)src0 + i3*nb03 + i2*nb02 + i1*nb01;
-    char * dst_ptr = (char *)dst + i3*nb3 + i2*nb2 + i1*nb1;
+    const char * dst_ptr = (const char *)dst + i3*nb3 + i2*nb2 + i1*nb1;
```

Comment:
```suggestion
    char * dst_ptr = (char *) dst + i3*nb3 + i2*nb2 + i1*nb1;
```

Casting `dst_ptr` to `const char *` and then writing to it afterwards is not correct.